tests/receivers/jmxreceiver/testdata/resource_metrics/all.yaml

```diff
@@ -0,0 +1,207 @@
+resource_metrics:
+  - attributes:
+      service.name: unknown_service:java
+      telemetry.sdk.language: java
+      telemetry.sdk.name: opentelemetry
+      telemetry.sdk.version: 1.7.0
```

Comment:
this will change with all updated versions so can we remove to not enforce the check?  the jmx metric gatherer integration tests take care of this for us.

Reply:
After removing this version attribute from the resource, the validation gets failed [here](https://github.com/rrupapara/splunk-otel-collector/blob/415f3a03290334e7af1687f6f34ad0ffa23a55cc/tests/testutils/resource_metrics.go#L358) due to tight equals. So what is your opinion if we can make it to relax equals as we do for [metric](https://github.com/rrupapara/splunk-otel-collector/blob/415f3a03290334e7af1687f6f34ad0ffa23a55cc/tests/testutils/resource_metrics.go#L236)? And also same needs to be applied for [instrumentationLibrary](https://github.com/rrupapara/splunk-otel-collector/blob/415f3a03290334e7af1687f6f34ad0ffa23a55cc/tests/testutils/resource_metrics.go#L202) validation.

Reply:
@rmfitzpatrick Could you please provide your feedback for the above comment?

Reply:
In lieu of relaxed equivalence helpers I'd use the resource processor and remove the `telemetry.sdk.version` attribute before exporting: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourceprocessor

Reply:
Removed `telemetry.sdk.version` using resourceprocessor. Would you mind providing any suggestion to also remove [this](https://github.com/rrupapara/splunk-otel-collector/blob/28335d786b91e5f17648d97d1b98d2bd3a4f41b4/tests/receivers/jmxreceiver/testdata/resource_metrics/all.yaml#L9) `version: 1.7.0` validation as well? As it seems to be not possible with any processors. Thanks!

Reply:
@rmfitzpatrick Could you please provide your input here?

Reply:
Changed the InstrumentationLibrary items validation from Equals to Relaxed Equal as per the discussion with Ryan, to skip the version validation from InstrumentationLibrary.